app/scripts/metamask-controller.js

```diff
@@ -1747,6 +1744,19 @@ export default class MetamaskController extends EventEmitter {
     };
   }
 
+  async getTokenStandardAndDetails(address, userAddress, tokenId) {
+    const details = await this.assetsContractController.getTokenStandardAndDetails(
+      address,
+      userAddress,
+      tokenId,
+    );
+    return {
+      ...details,
+      decimals: details?.decimals?.toString(10),
+      balance: details?.balance?.toString(10),
```

Comment:
`decimals` and `balance` are not converted from `BN`s to `string`s automagically in Firefox (as they are in Chrome) so we need to do this for now.